src/main/java/frc/robot/vision/SwitchableCamera.java

```diff
@@ -1,40 +1,90 @@
 package frc.robot.vision;
 
+import org.opencv.core.Core;
+import org.opencv.core.Mat;
+
 import edu.wpi.first.cameraserver.CameraServer;
+import edu.wpi.first.cscore.CvSink;
+import edu.wpi.first.cscore.CvSource;
 import edu.wpi.first.cscore.UsbCamera;
-import edu.wpi.first.cscore.VideoSink;
 import edu.wpi.first.cscore.VideoSource;
+import edu.wpi.first.wpilibj.DriverStation;
 
 /**
  * A `VideoSource` that toggles between a front and back camera.
  */
 public class SwitchableCamera {
-    private final UsbCamera front;
-    private final UsbCamera back;
-    private final VideoSink server;
+    private final UsbCamera topCam;
+    private final UsbCamera bottomCam;
+
+    private final int WIDTH = 140;
+    private final int HEIGHT = 120;
+
+    private final CvSink bottomSink;
+    private final CvSink topSink;
+
+    private final CvSource server;
+
+    private volatile boolean frontActive = true;
+
+    public SwitchableCamera() {
+        topCam = new UsbCamera("top cam", 0);
+        bottomCam = new UsbCamera("bot cam", 1);
+
+        topCam.setResolution(WIDTH, HEIGHT);
+        bottomCam.setResolution(WIDTH, HEIGHT);
+
+        topCam.setFPS(30);
+        bottomCam.setFPS(30);
+
+        bottomSink = new CvSink("flip bottom cam");
+        bottomSink.setSource(bottomCam);
+
+        topSink = new CvSink("flip bottom cam");
+        topSink.setSource(topCam);
+
+        this.server = CameraServer.putVideo("test", HEIGHT, WIDTH);
 
-    private boolean frontActive = true;
+        Runnable stream = () -> {
+            Mat img = new Mat();
+            while (!Thread.interrupted()) {
+                if (frontActive) {
+                    if (bottomSink.grabFrame(img) != 0) {
+                        Core.rotate(img, img, Core.ROTATE_90_CLOCKWISE);
+                        server.putFrame(img);
+                    } else {
+                        DriverStation.reportError(bottomSink.getError(), false);
+                    }
+                } else {
+                    if (topSink.grabFrame(img) != 0) {
```

Comment:
Maybe the `grabFrame` logic stuff can be abstracted into a method with a boolean for whether to rotate it?